src/components/views/messages/RoomCreate.tsx

```diff
@@ -37,31 +39,50 @@ interface IProps {
  * room.
  */
 export const RoomCreate: React.FC<IProps> = ({ mxEvent, timestamp }) => {
+    // Note: we ask the room for its predecessor here, instead of directly using
+    // the information inside mxEvent. This allows us the flexibility later to
+    // use a different predecessor (e.g. through MSC3946) and still display it
+    // in the timeline location of the create event.
+    const roomContext = useContext(RoomContext);
+    const predecessor = roomContext.room.findPredecessor();
```

Comment:
This doesn't look like it'll react to the predecessor changing due to a new/updated state event coming in?

Reply:
Addressed in 1b4c3f6745 . Apologies for the rebase - fat-fingered it.